### PR TITLE
test: Add unit test for RoomMenu options

### PR DESCRIPTION
**Pull request by csuadev**
This task add pending unit tests for the RoomMenu to validate a task related to Omnichannel conversations where we removed the option to hide a conversation when its an Omnichannel.

Related to this PR: https://github.com/RocketChat/Rocket.Chat/pull/32691

## Issue(s)
[https://rocketchat.atlassian.net/browse/CORE-579](https://rocketchat.atlassian.net/browse/CORE-579)
[https://rocketchat.atlassian.net/browse/SUP-599](https://rocketchat.atlassian.net/browse/SUP-599)

### CODE REVIEW

**Review by MarcosSpessatto — APPROVED** [review]

**Review by yash-rajpal — CHANGES_REQUESTED** [review]
Can we add more test cases, to cover more cases?

Example, only showing favorites option, based on the setting value?

**Review by yash-rajpal — DISMISSED** [review]
Lets move forward with this for now and we will add more cases in future.

File: apps/meteor/tests/unit/client/views/room/RoomMenu.spec.tsx
Comment (dougfabris) [inline apps/meteor/tests/unit/client/views/room/RoomMenu.spec.tsx:1]: let's try to mock using `mock-providers`

**Review by dougfabris — DISMISSED** [review]

File: apps/meteor/tests/unit/client/views/room/RoomMenu.spec.tsx
Comment (dougfabris) [reply apps/meteor/tests/unit/client/views/room/RoomMenu.spec.tsx:1]: The unit tests are broken

File: apps/meteor/client/sidebar/RoomMenu.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		})
+		.withSetting('Favorite_Rooms', true)
+		.withPermission('leave-c')
+		.withPermission('leave-p')
+		.build(),
+	legacyRoot: true,
+};
+
+describe('RoomMenu component', () => {
+	it('should display the menu options', async () => {
```
Comment (MarcosSpessatto) [inline apps/meteor/client/sidebar/RoomMenu.spec.tsx:47]: When should it be displayed? When it's a regular room? 🤔

File: apps/meteor/client/sidebar/RoomMenu.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		const menu = screen.queryByRole('button');
+		await userEvent.click(menu as HTMLElement);
+
+		expect(await screen.findByRole('option', { name: 'Hide' })).toBeInTheDocument();
+		expect(await screen.findByRole('option', { name: 'Favorite' })).toBeInTheDocument();
+		expect(await screen.findByRole('option', { name: 'Mark Unread' })).toBeInTheDocument();
+		expect(await screen.findByRole('option', { name: 'Leave' })).toBeInTheDocument();
+	});
+
+	it('should display the menu options mark unread and favorite', async () => {
```
Comment (MarcosSpessatto) [inline apps/meteor/client/sidebar/RoomMenu.spec.tsx:59]: An omnichannel room? 🤔

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

**Review by MarcosSpessatto — DISMISSED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]

**Pull request merged**